03b_model/src/run_model.py

```diff
@@ -54,19 +101,58 @@ def select_inputs_targets(inputs, target, train_start_date, test_end_date, out_d
     #inputs_df = inputs_df.join(nerrs_input)
     inputs_df = inputs_df[train_start_date:test_end_date]
     
+    if inc_ante:
+        
+        win_sizes = config['window_size']
+        ante_var = config['antecedant_variables']
+        
+        #if discharge is in the antecedant variable list make a combined
+        #discharge variable adding the discharge from the Delaware at Trenton (01474500)
+        #to the discharge from the Schuylkill at Philadelphia (01463500)
+        if 'discharge' in ante_var:
+            inputs_df['discharge'] = inputs_df['discharge_01463500']+inputs_df['discharge_01474500']
+        
+        for win_size in win_sizes:    
+            for j,col in enumerate(ante_var):
+                inputs_df[col+'_'+str(win_size).rjust(3,'0')+'_mean'] = float('nan')
+                col_values = []
+                for i in range(len(inputs_df)-win_size):
+                    temp_win = inputs_df[col].iloc[i:win_size+i].copy()
+                    win_mean = np.mean(temp_win)
+                    col_values.append(win_mean)
+                        
+                print('Calculating '+inputs_df.columns[-1]+' '+str(win_size))
+                inputs_df.iloc[win_size:len(inputs_df), -1] = col_values
+        #if discharge is in the antecedant variable list
+        if 'discharge' in ante_var:
+            #remove the combined discharge from the list of inputs
+            #this way we have the discharge at Trenton and from the Schuylkill as separate predictors
+            #and the antecedant discharge of the two combined as another predictor
+            inputs_df = inputs_df.drop(columns = 'discharge')
+        
+        inputs_df = inputs_df.dropna()
+    
     
     #read in the salt front record
     target_df = pd.read_csv(os.path.join('03a_it_analysis', 'in', 'saltfront.csv'), parse_dates = True, index_col = 'datetime')
-    target_df = target_df['saltfront_daily'].to_frame()
+    target_df = target_df['saltfront7_weekly'].to_frame()
```

Comment:
Would it help to make the variable to pull ('saltfront7_weekly') a function argument? Something like target_variable. Comment applies to all other functions in this code that specify target names.

Reply:
Agreed! I was waiting on the methods for calculating the salt front before fully integrating into the pipeline because I thought we might be reproducing it ourselves. I have changed it to a variable